### PR TITLE
Cloud migration: upload snapshot files using presigned url

### PR DESCRIPTION
**Pull request by PoorlyDefinedBehaviour**
Closes https://github.com/grafana/grafana-operator-experience-squad/issues/834
Closes https://github.com/grafana/grafana-operator-experience-squad/issues/842

### CODE REVIEW

File: pkg/services/cloudmigration/cloudmigrationimpl/snapshot_mgmt.go [hunk 1]
```diff
@@ -234,7 +236,7 @@ func (s *Service) buildSnapshot(ctx context.Context, signedInUser *user.SignedIn
 }
 
 // asynchronous process for and updating the snapshot status
-func (s *Service) uploadSnapshot(ctx context.Context, snapshotMeta cloudmigration.CloudMigrationSnapshot) {
+func (s *Service) uploadSnapshot(ctx context.Context, session *cloudmigration.CloudMigrationSession, snapshotMeta *cloudmigration.CloudMigrationSnapshot) (err error) {
```
Comment (mmandrus) [inline pkg/services/cloudmigration/cloudmigrationimpl/snapshot_mgmt.go:239]: nit: named return value isn't needed here

**Review by mmandrus — APPROVED** [review]
lgtm!

File: pkg/services/cloudmigration/cloudmigrationimpl/snapshot_mgmt.go (hunk 1 above)
Comment (mmandrus) [reply pkg/services/cloudmigration/cloudmigrationimpl/snapshot_mgmt.go:239]: I see why you did it upon further inspection... It's fine to leave I think

File: pkg/services/cloudmigration/cloudmigrationimpl/snapshot_mgmt.go [hunk 2]
```diff
@@ -247,34 +249,74 @@ func (s *Service) uploadSnapshot(ctx context.Context, snapshotMeta cloudmigratio
...
+	indexFilePath := filepath.Join(snapshotMeta.LocalDir, "index.json")
+	// LocalDir can be set in the configuration, therefore the file path can be set to any path.
+	// nolint:gosec
+	indexFile, err := os.Open(indexFilePath)
+	if err != nil {
+		return fmt.Errorf("opening index files: %w", err)
+	}
+	defer func() {
+		if closeErr := indexFile.Close(); closeErr != nil {
+			err = errors.Join(err, fmt.Errorf("closing index file: %w", closeErr))
```
Comment (mmandrus) [inline pkg/services/cloudmigration/cloudmigrationimpl/snapshot_mgmt.go:264]: @leandro-deveikis points out that we could potentially return an error if the snapshot is already uploaded. Maybe it is better to just log one instead?

Comment (PoorlyDefinedBehaviour) [reply pkg/services/cloudmigration/cloudmigrationimpl/snapshot_mgmt.go]: Changed to a log statement

Comment (mmandrus) [reply pkg/services/cloudmigration/cloudmigrationimpl/snapshot_mgmt.go]: LGTM

**Pull request merged**